src/deadline_worker_agent/sessions/session.py

```diff
@@ -242,6 +247,13 @@ def os_user(self) -> Optional[SessionUser]:
         """The session user"""
         return self._os_user
 
+    @property
+    def manifest_paths_by_root(self) -> dict[str, str]:
+        return self._manifest_paths_by_root
+
+    def set_manifest_paths_by_root(self, manifest_paths_by_root) -> None:
+        self._manifest_paths_by_root = manifest_paths_by_root
```

Comment:
Two things:

1.  nit: we could use a [property setter](https://docs.python.org/3/library/functions.html#property) here
2. we should probably make a copy rather than setting a reference

Putting this together:

```suggestion
    @manifest_paths_by_root.setter
    def manifest_paths_by_root(self, value: dict[str, str]) -> None:
        self._manifest_paths_by_root = dict(value)
```

Reply:
Good catch, will incorporate, thanks!